### PR TITLE
[RHDEVDOCS-1557] Adding and removing services in context of the application

### PR DESCRIPTION
**Pull request by rishumehra**
**Purpose**:

- Description of how to use the Context Menu in the topology **Graph view**
- Using context menu to add/remove services in the application group

**PR information**:

- Aligned team: DevConsole
- For branches: OpenShift 4.8 Async, OpenShift 4.9
- Jira: https://issues.redhat.com/browse/RHDEVDOCS-1557
- Direct link to doc preview: https://deploy-preview-35180--osdocs.netlify.app/openshift-enterprise/latest/applications/application_life_cycle_management/odc-viewing-application-composition-using-topology-view.html#odc-adding-services-to-your-application_viewing-application-composition-using-topology-view
- SME review: @karthikjeeyar
- QE review: @makambalaji
- Peer review: @rolfedh

### CODE REVIEW

File: modules/odc-removing-services-to-application.adoc [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+[id="odc-removing-services-to-your-application_{context}"]
+= Removing services to your application
+
+In the *Topology view*, remove a service from your application using the context menu or sidebar in your application.
+
+.Procedure
+
+To remove a service using the context menu:
+
+1. Right-click on a service in an application group in the *Topology* view to see the context menu.
```
Comment (makambalaji) [inline modules/odc-removing-services-to-application.adoc:14]:
```suggestion
1. Right-click on a service in an application group in the *Topology* graph view to see the context menu.
```

File: modules/odc-adding-services-to-application.adoc [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+[id="odc-adding-services-to-your-application_{context}"]
+= Adding services to your application
+
+In the *Topology view*, access the *+Add* actions using the context menu, sidebar, or by hovering and dragging the dangling arrow on to the *Topology view* to add a service to your application.
+
+.Procedure
+
+To add a service using the context menu:
+
+1. Right-click inside an application group in the *Topology* view  to see the context menu.
```
Comment (makambalaji) [inline modules/odc-adding-services-to-application.adoc:14]:
```suggestion
1. Right-click inside an application group in the *Topology* graph view  to see the context menu.
```

File: modules/odc-adding-services-to-application.adoc [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// Module included in the following assemblies:
+//
+// applications/application_life_cycle_management/odc-viewing-application-composition-using-topology-view.adoc
+
+[id="odc-adding-services-to-your-application_{context}"]
+= Adding services to your application
+
+In the *Topology view*, access the *+Add* actions using the context menu, sidebar, or by hovering and dragging the dangling arrow on to the *Topology view* to add a service to your application.
```
Comment (rolfedh) [inline modules/odc-adding-services-to-application.adoc:8]:
```suggestion
To add a service to your application, in the *Topology view*, access the *+Add* actions using the context menu, sidebar, or by hovering and dragging the dangling arrow on to the *Topology view*.
```

Comment (rolfedh) [reply modules/odc-adding-services-to-application.adoc:8]: Topology graph?

File: modules/odc-adding-services-to-application.adoc [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+.Procedure
+
+To add a service using the context menu:
```
Comment (rolfedh) [inline modules/odc-adding-services-to-application.adoc:12]:
```suggestion
```

File: modules/odc-adding-services-to-application.adoc (hunk 3 above)
Comment (rolfedh) [reply modules/odc-adding-services-to-application.adoc:8]: Is it better to keep this abstract lean and not mention the different UI methods here? (`context menu, sidebar, or by hovering and dragging the dangling arrow` ) 

The procedure focuses on one specific UI method, the context menu. If the other methods are important for users to know about, you could mention them in a note. For example,
```
[NOTE]
====
In addition to the context menu, you can add services by using the sidebar or hovering and dragging the dangling arrow to the Topology graph
====
```

File: modules/odc-adding-services-to-application.adoc (hunk 4 above)
Comment (rolfedh) [reply modules/odc-adding-services-to-application.adoc:12]: The IBM Style Guide or Red Hat SSG recommends avoiding this type of sentence. The information it contains is redundant with the title and the contents of the procedure.

File: modules/odc-adding-services-to-application.adoc (hunk 2 above)
Comment (rolfedh) [reply modules/odc-adding-services-to-application.adoc:14]:
```suggestion
. Right-click an application group in the *Topology* graph.
```

File: modules/odc-adding-services-to-application.adoc [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+1. Right-click inside an application group in the *Topology* view  to see the context menu.
++
+.Add resource context menu
+image::odc_context_menu.png[]
+
+2. Select *Add to Application*, then select the required method *From Git*, *Container Image*, *From Dockerfile*, *From Devfile*, *Upload JAR file*, *Event Source*, *Channel*, and *Broker* to add a service to the application group. For example,
```
Comment (rolfedh) [inline modules/odc-adding-services-to-application.adoc:19]:
```suggestion
. Select a method for adding a service to the application group, such as *From Git*, *Container Image*, *From Dockerfile*, *From Devfile*, *Upload JAR file*, *Event Source*, *Channel*, or *Broker* and complete the required information. For example:
```

File: modules/odc-adding-services-to-application.adoc [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+** Select *From Git* in the context menu to see the *Import from Git* form.
```
Comment (rolfedh) [inline modules/odc-adding-services-to-application.adoc:20]:
```suggestion
** Complete the form for the method you choose. For example, for the *From Git* method, complete the *Import from Git* form.
```

File: modules/odc-adding-services-to-application.adoc [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+** Fill in the required details and click *Create*, to add a service based on the source code in your Git repository.
```
Comment (rolfedh) [inline modules/odc-adding-services-to-application.adoc:21]:
```suggestion

For example, to add a service based on the source code in your Git repository, choose the *From Git* method, fill in the *Import from Git* form, and click *Create*.
```

File: modules/odc-removing-services-to-application.adoc [hunk 8]
```diff
@@ -0,0 +1,19 @@
+// Module included in the following assemblies:
+//
+// applications/application_life_cycle_management/odc-viewing-application-composition-using-topology-view.adoc
+
+[id="odc-removing-services-to-your-application_{context}"]
+= Removing services to your application
```
Comment (rolfedh) [inline modules/odc-removing-services-to-application.adoc:6]: You can extend the suggestions you choose to apply from the previous topic to this one.

Comment (rolfedh) [reply modules/odc-removing-services-to-application.adoc:6]:
```suggestion
= Removing services from your application
```

**Pull request merged**